### DISCUSSION TITLE
bugs  it ask for root

### DISCUSSION THREAD

**Discussion by SergeyMy** [rustdesk/rustdesk#4273] [General]
why does it ask for root permissions when unlocking settings?
![изображение](https://user-images.githubusercontent.com/131688106/236264399-68ba5b47-ca6e-4c85-8cf6-3b7325ed2c81.png)

(EL8) linux desktop, flutter, 357028b2 ("Merge pull request #4271 from rustdesk/revert-4265-lpcy-touch-pad", 2023-05-04)
![изображение](https://user-images.githubusercontent.com/131688106/236265087-05b6ac9b-d559-43b9-a801-59b009fe8e27.png)
where to fix it break?

**Comment by SergeyMy** [comment]
why would an ordinary user need **root** rights to change the settings in their **personal profile** ~/.config/rustdesk?

_flutter/lib/desktop/pages/desktop_setting_page.dart_
1875 bool checked = await bind.mainCheck**SuperUserPermission**();

**Comment by rustdesk** [comment]
Good question, we will remove from ~/.config

**Comment by SergeyMy** [comment]
hmm... rustdeskt on linux desktop is often a client and not a server, linux server has many other ways of remote administration without a local UI.
do not break the desktop version, please. 

PS
maybe? :)
![изображение](https://user-images.githubusercontent.com/131688106/236419962-85ddb6de-cc32-4cce-a73b-774901a88392.png)
[bugs_it_ask_for_root_4273.diff.gz](https://github.com/rustdesk/rustdesk/files/11404838/bugs_it_ask_for_root_4273.diff.gz)

**Comment by SergeyMy** [comment]
user setting: ./rustdesk
service setup: sudo ./rustdesk
in both cases the configuration in "$HOME/.config/rustdesk 
:) getSuperUserPermission doesn't make sense!

PS
configuration in "$HOME/.config/rustdesk = portable version!

**Comment by SergeyMy** [comment]
@interesting way to test... :(
1. service started as root
2. run rustdesk from anywhere other than /usr
bingo we change the service settings without any rights!.

bool checked = !bind.**mainIsInstalled**

src/bridge_generated.rs

src/platform/linux.rs
pub fn is_installed() -> bool {
    if let Ok(p) = std::env::current_exe() {
        p.to_str().unwrap_or_default().**starts_with("/usr")**
    } else {
        false
    }
}

**Comment by gagero** [comment]
This could be easily fixed with a call to `id -u` and checking if the output is 0 or not.

**Comment by SergeyMy** [comment]
the question is not how to check, but how to **restrict write access** to /tmp/RustDesk/ipc*

for myself , I did this:
![изображение](https://github.com/rustdesk/rustdesk/assets/131688106/ebd4f3b7-3cc3-4878-a6cf-0fde8ced0b71)

//a user without rights will simply **overwrite the socket name** :):):)
